app/components/buttons/EnableExternalRequestButton.js

```diff
@@ -2,11 +2,11 @@ import { settings } from "connectors";
 import KeyBlueButton from "./KeyBlueButton";
 
 const EnableExternalRequestButton = ({ children, requestType,
-  onAddAllowedRequestType }) => {
+  onAddAllowedRequestType, onClick }) => {
 
-  const onClick = () => onAddAllowedRequestType(requestType);
+  const onClickFunction = () => onClick ? onClick() && onAddAllowedRequestType(requestType) : onAddAllowedRequestType(requestType);
```

Comment:
Shouldn't this be the other way around (`onAddAllowedRequestType(requestType) && onClick`)? So that you're guaranteed that when `onClick()` is triggered the new request type will be allowed to be done?

Otherwise you might end up with a race where onClick() tries to execute a given request but that one isn't allowed yet.

Reply:
yeah, you are right. 

I fixed it as well